Sources/Networking/HTTPClient.swift

```diff
@@ -355,3 +377,18 @@ private extension Encodable {
     }
 
 }
+
+private extension Result where Success == HTTPResponse<Data>, Failure == Error {
+
+    // Parses a `Result<HTTPResponse<Data>>` to `Result<HTTPResponse<Value>>`
+    func parseResponse<Value: HTTPResponseBody>() -> HTTPResponse<Value>.Result {
+        return self.flatMap { response in
+            HTTPResponse<Value>.Result {
+                try response.mapBody { data in
+                    try Value.create(with: data)
+                }
+            }
+        }
```

Comment:
I understand this code might be hard to follow, it's peak "functional" code. Ultimately it's just converting from `Data` to `Value` by calling the `HTTPResponseBody` decoding method. The nesting is because it's inside of the hierarchy of types of `Result` and `HTTPResponse`.

Feedback welcome if there's objections about it.

Reply:
> Honestly, I had a tough time understanding this. I think the mixture of functional methods, generic types, conversion of result types, Result.init with the block passed in directly, deep nesting, and implicit returns was a bit too much for my brain to handle.

Totally fair.

How about explaining what each line does in English (plus maybe adding all explicit types)?

```swift
// Parses a `Result<HTTPResponse<Data>>` to `Result<HTTPResponse<Value>>`
func parseResponse<Value: HTTPResponseBody>() -> HTTPResponse<Value>.Result {
    return self.flatMap { response in          // Convert the `Result` type
        HTTPResponse<Value>.Result {           // Create a new `HTTPResponse<Value>`
            try response.mapBody { data in     // Convert the body of `HTTPResponse<Data>` from `Data` -> `Value`
                try Value.create(with: data)   // Decode `Data` into `Value`
            }
        }
    }
}
```

Reply:
Something to comment on this, is that this code is doing just a data transformation, it's not like it has some weird logic that needs to be maintained. There really is only one way to implement this, maybe many ways of _writing it_, but they would all compile to the same thing.